src/common/transformations/src/transformations/common_optimizations/convert_u4_weights_zero_point_to_scalar.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/convert_u4_weights_zero_point_to_scalar.hpp"
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/reference/autobroadcast_binop.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace {
+bool convert_zero_point_to_scalar(const std::shared_ptr<ov::op::v0::Constant>& weights,
+                                  const std::shared_ptr<ov::op::v0::Constant>& zero_point) {
+    auto zero_point_shape = zero_point->get_shape();
+    if (ov::shape_size(zero_point_shape) == 1)
+        return false;
+
+    const auto& weights_shape = weights->get_shape();
+    const size_t weights_rank = weights_shape.size();
+    const size_t zero_point_rank = zero_point_shape.size();
+    // Zero point constant can be converted into scalar only if this does not affect Subtract output shape
+    if (weights_rank < zero_point_rank || ov::shape_size(weights_shape) < ov::shape_size(zero_point_shape))
+        return false;
+
+    zero_point_shape.insert(zero_point_shape.begin(), weights_rank - zero_point_rank, 1);
+    for (size_t i = 0; i < weights_rank; ++i) {
+        if (zero_point_shape[i] > weights_shape[i])
+            return false;
+    }
+
+    int8_t zero_point_value = zero_point->cast_vector<int8_t>(1)[0];
+    if (!ov::op::util::constantIsEqualTo(zero_point, static_cast<float>(zero_point_value)))
+        return false;
+
+    const auto new_zp = ov::op::v0::Constant::create(zero_point->get_element_type(), {}, {zero_point_value});
```

Comment:
As the check at the beginning of the function assures the data in zero_point is not a scalar, why can we just use the first value of zero_point (i.e. {zero_point_value}) to create new_zp? Did I miss anything?

Reply:
At L36 we use `ov::op::util::constantIsEqualTo` which checks that all values in `zero_point` are equal to `zero_point_value`. If any value is not equal to `zero_point_value`, this function returns false and `new_zp` is not created. So we create `new_zp` only in valid case